### ISSUE TITLE
Count and Exists projection for @Query

### ISSUE THREAD

**Issue by lihanyu23** [arangodb/spring-data#21]
Feature request:

Add options for countProjection and existsProjection in @Query.

For the moment, results are only converted by ArangoResultConverter with predefined type mappings.
While there is a need for writing queries that count or check existence just like derived queries.

**Comment by christian-lechner** [comment]
Can't you just do the following:
```java
// Count for collection
@Query("RETURN COUNT(@@collection)")
// Count for a query
@Query("RETURN COUNT(FOR doc IN ... FILTER ... RETURN doc)")

// Exists for a specific document
@Query("RETURN DOCUMENT(@id) != NULL")
// Exists for a query
@Query("RETURN COUNT(FOR doc IN ... FILTER ... RETURN doc) > 0")
```

**Comment by mpv1989** [comment]
Fixed in 2.1.0
